### PR TITLE
2.x - MicroProfile 4 and Vert.x 4

### LINKED ISSUE #13161: Using Quarkus Redis Client to connect to SSL enabled Redis server throws an error
**Describe the bug**
Can't connect to Redis server using the SSL connection.

**Expected behavior**
Established connection and ability to PING Redis server.

**Actual behavior**
Exception while trying to PING Redis server:
```
java.util.concurrent.CompletionException: CONNECTION_CLOSED
	at io.smallrye.mutiny.operators.UniBlockingAwait.await(UniBlockingAwait.java:72)
	at io.smallrye.mutiny.groups.UniAwait.atMost(UniAwait.java:61)
	at io.quarkus.redis.client.runtime.RedisClientImpl.await(RedisClientImpl.java:1026)
	at io.quarkus.redis.client.runtime.RedisClientImpl.ping(RedisClientImpl.java:517)
```

**To Reproduce**

1. run Redis server with SSL enabled
2. configure Quarkus Redis Client to connect to SSL enabled Redis server
3. run `mvn quarkus:dev`
3. try to ping the server

This is how I inject and use Redis client:
```
    @Inject
    RedisClient redisClient;
    ...

    redisClient.ping(Collections.emptyList()); // this call throws an exception
```

**Configuration**
```
%dev.quarkus.redis.hosts=rediss://default:<password>@<hostname>:<port>
```

**Additional context**
I don't see any configuration parameters related to SSL here:
https://quarkus.io/guides/redis#quarkus-redis-client_configuration

But looking into code, it's obvious that there is a support for `rediss` scheme, indicating that SSL should be supported. Am I missing some certificate configuration here?

Just to be clear, I've tried connecting to the same Redis server using `redli` (Redis client with built-in SSL support, https://github.com/IBM-Cloud/redli/):
```
$ redli --uri rediss://default:<password>@<hostname>:<port>
Connected to 6.0.9
> ping
PONG
>
```
In other words, no problems, everything worked as expected.

Also, I've then tried SSH tunneling:
```
$ cat /etc/stunnel/redis.conf
pid = /run/stunnel-redis.pid

[redis-client]
client = yes
accept = 127.0.0.1:6800
connect = <hostname>:<port>

$ sudo stunnel /etc/stunnel/redis.conf
```
After creating the tunnel and running the Quarkus app again, with reconfigured property:
```
%dev.quarkus.redis.hosts=redis://default:<password>@localhost:6800
```
the 
```
   redisClient.ping(Collections.emptyList());
```
call worked as expected, i.e it successfully connected to SSL enabled Redis server using the SSH tunnel.

I've tried both Quarkus `1.9.1.Final` and `1.9.2.Final`.

I've even tried configuring the Redis Client using these obsolete properties:
```
%dev.quarkus.redis.hosts=<hostname>:<port>
%dev.quarkus.redis.password=<password>
%dev.quarkus.redis.ssl=true
```
but didn't have any luck with it.

Any thoughts?

I'm I missing something or there really is a bug?

### LINKED ISSUE #14532: Redis Sentinel not working with Quarkus 1.11.2
**Describe the bug**
Hello. Quarkus is not being able to connect with Redis Sentinel. The standalone option works as expected. The issue is reproducible with both Quarkus 1.10.5 and 1.11.0

**To Reproduce**

You can use the following project to reproduce the issue: https://github.com/akoufa/quarkus_sentinel_issue. In this project I am configuring two Redis clients one with sentinel and one standalone.

1. Run `docker-compose up -d` and then start Quarkus Server
2. Visit http://localhost:8080/hello/standalone works as expected
3. Visit http://localhost:8080/hello/sentinel throws error

**Configuration**
```
quarkus.redis.client-type=sentinel
quarkus.redis.hosts=redis://localhost:26379

quarkus.redis.standalone.client-type=standalone
quarkus.redis.standalone.hosts=redis://localhost:6379

```

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

**Comment by gsmet** [comment]
`TooManyLabelsOverflow`.

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisConfig.java [hunk 1]
```diff
@@ -90,10 +90,10 @@
         public Optional<RedisRole> role;
 
         /**
-         * Whether or not to use slave nodes (only considered in Cluster mode).
+         * Whether or not to use replicas nodes (only considered in Cluster mode).
          */
         @ConfigItem(defaultValueDocumentation = "never")
-        public Optional<RedisSlaves> slaves;
+        public Optional<RedisReplicas> replicas;
```
Comment (machi1990) [inline extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisConfig.java:96]: Love the change.

**Review by gsmet — CHANGES_REQUESTED** [review]
Please don't merge, we need to coordinate this thing.

File: integration-tests/reactive-messaging-kafka/pom.xml [hunk 2]
```diff
@@ -13,6 +13,12 @@
     <name>Quarkus - Integration Tests - Reactive Messaging - Kafka</name>
     <description>The Kafka Reactive Messaging integration tests module</description>
 
+    <properties>
+        <maven.compiler.parameters>true</maven.compiler.parameters>
+        <!-- needs to wait for smallrye-reactive-messaging-health to update MP health to 3.0 -->
+        <enforcer.skip>true</enforcer.skip>
```
Comment (famod) [inline integration-tests/reactive-messaging-kafka/pom.xml:19]: Is this about `dependencyConvergence`? I'm a bit wary about skipping _all_ enforcer rules like this, e.g. see #15841.
I understand that it's of utmost importance get this branch merged but this should be narrowed ASAP after the merge.

Comment (cescoffier) [reply integration-tests/reactive-messaging-kafka/pom.xml:19]: I will fix it just after the merge. 
MP Health has nasty breaking changes, so we had to work upstream first. I've all fine now.

**Review by gsmet — APPROVED** [review]

**Comment by geoand** [comment]
What a milestone!

AWESOME work @cescoffier!

**Comment by machi1990** [comment]
Great work! @cescoffier

**Pull request merged**